### PR TITLE
Consistent log prefixes and messages

### PR DESCRIPTION
**Pull request by ivkos**
Throughout the library there were log npmlog calls with a missing prefix (i.e. passing the message as the prefix). Fixed that.

Also removed redundant "Error in someFunction" in the prefix/message since this information is present in the function call (e.g. log.error, log.warn)

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -575,9 +575,9 @@ function makeDefaults(html, userID) {
 function parseAndCheckLogin(jar, defaultFuncs) {
   return function(data) {
     return bluebird.try(function() {
-      log.verbose("parseAndCheckLogin: " + data.body);
+      log.verbose("parseAndCheckLogin", data.body);
```
Comment (ivkos) [inline utils.js]: This may (or may not) as well fix a memory leak I'm currently investigating, as seen here:

**Comment by ivkos** [comment]
Apart from the high memory usage (essentially doubled) caused by v8 storing both:
* `"{ long 700k string }"`
* `"parseAndCheckLogin: { long 700k string }"`

which got fixed by the first commit in the PR, it turned out the `npmlog` module also [stores a record](https://github.com/npm/npmlog/blob/v1.2.1/log.js#L161) of all log messages independent of the logLevel setting. There's an option `log.maxRecordSize` ([defaults to 10000](https://github.com/npm/npmlog/blob/v1.2.1/log.js#L129)). Exposing this to `api.setOptions()` might be useful since storing 10k*700k strings hugely increases memory usage (> 7 GB).

**Comment by Schmavery** [comment]
Wow! Thanks so much for doing this.  We've been aware of some memory issues for a while and never had the chance to diver deeper into them.  The api also sorely needed some logging cleanup too 😄.
❤️ ❤️

**Comment by Schmavery** [comment]
Is this ready to be merged?

**Comment by ivkos** [comment]
Sure! 👍

File: src/getThreadList.js [hunk 2]
```diff
@@ -52,11 +52,11 @@ module.exports = function(defaultFuncs, api, ctx) {
         if (resData.error) {
           throw resData;
         }
-        log.verbose("Response in getThreadList: " + JSON.stringify(resData.payload.threads));
+        log.verbose("getThreadList", JSON.stringify(resData.payload.threads));
```
Comment (bsansouci) [inline src/getThreadList.js]: We could remove this `JSON.stringify` here as we're not concatenating two strings anymore (I'm sure the logger supports printing objs).

**Comment by bsansouci** [comment]
Thanks a loooot for that @ivkos! Here's some hearts for you ❤️ ❤️ ❤️

**Pull request merged**